### PR TITLE
feat: new onServerError hook triggered on server errors

### PR DESCRIPTION
**Pull request by vripoche**
Hi, 

Due to the unability of add error middleware in `onCreateServer` hook because the hook call is before SSR middleware, I propose to add a new node hook to manage SSR errors in project plugins: `onServerError`.
It is called on each server error, with error (and other hook parameters as `config`) in first argument. It also takes the plugin config as a second parameter if it needed.

### CODE REVIEW

**Comment by neoziro** [comment]
Hello @vripoche, what is the exact use-case? You can't do it in the React world?

**Comment by vripoche** [comment]
Hello Greg,

My use case is to log errors to Sentry through the node module (SSR side):
https://www.npmjs.com/package/@sentry/node

I tried to do that in `_errror.js` or with ErrorBoundaries in `_app.js` but the module cannot be loaded in browser, and that files are loaded both in SSR and in browser.

If I read the Express documentation I should add an error middleware, and I should add it in `onCreateServer` hook, but it is not working.

**Comment by neoziro** [comment]
@vripoche OK I understand the point. I will merge it as soon as I have time.

**Comment by vripoche** [comment]
Hello @neoziro,

Did you have enough time to review this PR ? ;)

Thanks for your attention.

File: packages/smooth/src/server/expressApp.js [hunk 1]
```diff
@@ -66,6 +66,11 @@ export function createExpressApp({
   apolloServer.applyMiddleware({ app })
 
   app.use(ssr({ config, schema, fragmentTypes }))
+  
+  app.use((error, req, res, next) => {
+    onServerError(config)({ error })
```
Comment (neoziro) [inline packages/smooth/src/server/expressApp.js]: Please pass `req`, it could be useful.

Comment (vripoche) [reply packages/smooth/src/server/expressApp.js]: Done

File: packages/smooth/src/plugin/nodeHooks/onServerError.js [hunk 2]
```diff
@@ -0,0 +1,4 @@
+import { applyHook } from '../node'
+
+export const onServerError = config => ({ error }) =>
```
Comment (neoziro) [inline packages/smooth/src/plugin/nodeHooks/onServerError.js]: req ?

**Pull request merged**